src/main/java/com/training/assignments/Operations.java

```diff
@@ -57,39 +76,38 @@ public String getPlaceBehaviourBased(String people) {
         } else if (people == "Very Bad") {
             return "go to hell";
         }
-        return " ";
+        return "";
     }
 
     public int getMiddleNumber(int number) {
         int index;
         for ( index = 1; index < number; index++) {
             if (number / index == number/2) {
                 logger.info("value of middle number is-->" + number / index);
-                break;
+                return number / index;
             }
         }
-        return number / index;
+        return 0;
     }
 
-    public int[] getEvenNumbers(int limit){
-        int index=1;
+    public int[] getEvenNumbers(int fromIndex,int toIndex){
         int array[]=new int[50];
         int number=0;
-        while(index <=limit){
-            if(index%2==0){
-                logger.info("the even number between 1 to 100 is-->"+index);
-                array[number]=index;
+        while(fromIndex <=toIndex){
+            if(fromIndex%2==0){
+                logger.info("the even number between 1 to 100 is-->"+fromIndex);
```

Comment:
Remove this logger, because the range is between 1-100, Console will be full of loggers, if you get a runtime or compile-time, then it will be difficult to navigate.